packages/plugins/typescript/typescript/src/visitor.ts

```diff
@@ -123,6 +128,39 @@ export class TsVisitor<
     return `Maybe<${super.ListType(node)}>`;
   }
 
+  UnionTypeDefinition(node: UnionTypeDefinitionNode, key: string | number | undefined, parent: any): string {
+    if (this.config.onlyOperationTypes) return '';
+    let withFutureAddedValue: string[] = [];
+    if (this.config.futureProofUnions) {
+      if (this.config.addTypename) {
+        if (this.config.nonOptionalTypename) {
```

Comment:
Maybe you can de-duplicate this? If `this.config.nonOptionalTypename` only effects the `?` sign, it can be inlined.

Reply:
We don't need this anymore.